// viser-ng/__ivy_ngcc__/es/plugins/Slider.js
import * as ɵngcc0 from '@angular/core';
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input } from '@angular/core';
import { PluginComponent } from './Plugin';
var Slider = (function (_super) {
    __extends(Slider, _super);
    function Slider() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "xAxis");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "yAxis");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "data");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "container");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "width");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "height");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "padding");
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], Slider.prototype, "start");
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], Slider.prototype, "end");
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], Slider.prototype, "minSpan");
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], Slider.prototype, "maxSpan");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "scales");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "fillerStyle");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "backgroundStyle");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "textStyle");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "handleStyle");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "backgroundChart");
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], Slider.prototype, "onChange");
Slider.ɵfac = function Slider_Factory(t) { return ɵSlider_BaseFactory(t || Slider); };
Slider.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: Slider, selectors: [["v-slider"]], inputs: { xAxis: "xAxis", yAxis: "yAxis", data: "data", container: "container", width: "width", height: "height", padding: "padding", start: "start", end: "end", minSpan: "minSpan", maxSpan: "maxSpan", scales: "scales", fillerStyle: "fillerStyle", backgroundStyle: "backgroundStyle", textStyle: "textStyle", handleStyle: "handleStyle", backgroundChart: "backgroundChart", onChange: "onChange" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [[3, "id"]], template: function Slider_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵpropertyInterpolate("id", ctx.context.container);
    } }, encapsulation: 2 });
var ɵSlider_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(Slider);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(Slider, [{
        type: Component,
        args: [{
                selector: 'v-slider',
                template: "<div id=\"{{context.container}}\"></div>"
            }]
    }], null, { xAxis: [{
            type: Input
        }], yAxis: [{
            type: Input
        }], data: [{
            type: Input
        }], container: [{
            type: Input
        }], width: [{
            type: Input
        }], height: [{
            type: Input
        }], padding: [{
            type: Input
        }], start: [{
            type: Input
        }], end: [{
            type: Input
        }], minSpan: [{
            type: Input
        }], maxSpan: [{
            type: Input
        }], scales: [{
            type: Input
        }], fillerStyle: [{
            type: Input
        }], backgroundStyle: [{
            type: Input
        }], textStyle: [{
            type: Input
        }], handleStyle: [{
            type: Input
        }], backgroundChart: [{
            type: Input
        }], onChange: [{
            type: Input
        }] }); })();
    return Slider;
}(PluginComponent));
export { Slider };
