// viser-ng/__ivy_ngcc__/es/module.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import * as viser from 'viser';
import { Chart } from './Chart';
import { Area, Axis, Bar, Box, Brush, Candle, Contour, Coord, DashLine, DodgeBar, DodgeInterval, Edge, ErrorBar, Facet, FacetView, Funnel, Guide, Heatmap, Interval, JitterPoint, Legend, Line, Path, Pie, Point, Polygon, Pyramid, Sankey, Schema, Sector, Series, SmoothArea, SmoothLine, StackArea, StackBar, StackInterval, Tooltip, Venn, View } from './components/index';
import { LiteChart } from './LiteChart';
import { PluginComponent } from './plugins/Plugin';
import { Slider } from './plugins/Slider';
var ViserModule = (function () {
    function ViserModule() {
    }
ViserModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ViserModule });
ViserModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ViserModule_Factory(t) { return new (t || ViserModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ViserModule, { declarations: function () { return [Chart,
        LiteChart,
        Axis,
        Brush,
        Coord,
        Facet,
        Guide,
        Legend,
        Tooltip,
        View,
        FacetView,
        Series,
        Pie,
        Sector,
        Line,
        SmoothLine,
        DashLine,
        Area,
        StackArea,
        SmoothArea,
        Bar,
        StackBar,
        DodgeBar,
        Point,
        Funnel,
        Pyramid,
        Schema,
        Box,
        Candle,
        Polygon,
        Contour,
        Heatmap,
        Edge,
        Sankey,
        ErrorBar,
        JitterPoint,
        StackInterval,
        DodgeInterval,
        Interval,
        Path,
        Venn,
        PluginComponent,
        Slider]; }, imports: function () { return [CommonModule]; }, exports: function () { return [Chart,
        LiteChart,
        Axis,
        Brush,
        Coord,
        Facet,
        Guide,
        Legend,
        Tooltip,
        View,
        FacetView,
        Series,
        Pie,
        Sector,
        Line,
        SmoothLine,
        DashLine,
        Area,
        StackArea,
        SmoothArea,
        Bar,
        StackBar,
        DodgeBar,
        Point,
        Funnel,
        Pyramid,
        Schema,
        Box,
        Candle,
        Polygon,
        Contour,
        Heatmap,
        Edge,
        Sankey,
        ErrorBar,
        JitterPoint,
        StackInterval,
        DodgeInterval,
        Interval,
        Path,
        Venn,
        PluginComponent,
        Slider]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ViserModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [
                    Chart,
                    LiteChart,
                    Axis,
                    Brush,
                    Coord,
                    Facet,
                    Guide,
                    Legend,
                    Tooltip,
                    View,
                    FacetView,
                    Series,
                    Pie,
                    Sector,
                    Line,
                    SmoothLine,
                    DashLine,
                    Area,
                    StackArea,
                    SmoothArea,
                    Bar,
                    StackBar,
                    DodgeBar,
                    Point,
                    Funnel,
                    Pyramid,
                    Schema,
                    Box,
                    Candle,
                    Polygon,
                    Contour,
                    Heatmap,
                    Edge,
                    Sankey,
                    ErrorBar,
                    JitterPoint,
                    StackInterval,
                    DodgeInterval,
                    Interval,
                    Path,
                    Venn,
                    PluginComponent,
                    Slider,
                ],
                exports: [
                    Chart,
                    LiteChart,
                    Axis,
                    Brush,
                    Coord,
                    Facet,
                    Guide,
                    Legend,
                    Tooltip,
                    View,
                    FacetView,
                    Series,
                    Pie,
                    Sector,
                    Line,
                    SmoothLine,
                    DashLine,
                    Area,
                    StackArea,
                    SmoothArea,
                    Bar,
                    StackBar,
                    DodgeBar,
                    Point,
                    Funnel,
                    Pyramid,
                    Schema,
                    Box,
                    Candle,
                    Polygon,
                    Contour,
                    Heatmap,
                    Edge,
                    Sankey,
                    ErrorBar,
                    JitterPoint,
                    StackInterval,
                    DodgeInterval,
                    Interval,
                    Path,
                    Venn,
                    PluginComponent,
                    Slider,
                ]
            }]
    }], function () { return []; }, null); })();
ɵngcc0.ɵɵsetComponentScope(PluginComponent, [ɵngcc1.NgClass, ɵngcc1.NgComponentOutlet, ɵngcc1.NgForOf, ɵngcc1.NgIf, ɵngcc1.NgTemplateOutlet, ɵngcc1.NgStyle, ɵngcc1.NgSwitch, ɵngcc1.NgSwitchCase, ɵngcc1.NgSwitchDefault, ɵngcc1.NgPlural, ɵngcc1.NgPluralCase, Chart,
    LiteChart,
    Axis,
    Brush,
    Coord,
    Facet,
    Guide,
    Legend,
    Tooltip,
    View,
    FacetView,
    Series,
    Pie,
    Sector,
    Line,
    SmoothLine,
    DashLine,
    Area,
    StackArea,
    SmoothArea,
    Bar,
    StackBar,
    DodgeBar,
    Point,
    Funnel,
    Pyramid,
    Schema,
    Box,
    Candle,
    Polygon,
    Contour,
    Heatmap,
    Edge,
    Sankey,
    ErrorBar,
    JitterPoint,
    StackInterval,
    DodgeInterval,
    Interval,
    Path,
    Venn,
    PluginComponent,
    Slider], [ɵngcc1.AsyncPipe, ɵngcc1.UpperCasePipe, ɵngcc1.LowerCasePipe, ɵngcc1.JsonPipe, ɵngcc1.SlicePipe, ɵngcc1.DecimalPipe, ɵngcc1.PercentPipe, ɵngcc1.TitleCasePipe, ɵngcc1.CurrencyPipe, ɵngcc1.DatePipe, ɵngcc1.I18nPluralPipe, ɵngcc1.I18nSelectPipe, ɵngcc1.KeyValuePipe]);
    return ViserModule;
}());
export { ViserModule };
export var registerAnimation = viser.registerAnimation;
export var registerShape = viser.registerShape;
export var Global = viser.Global;
